cpp/src/Ice/ConnectionI.cpp

```diff
@@ -605,72 +623,6 @@ Ice::ConnectionI::updateObserver()
     _observer.attach(o);
 }
 
-void
-Ice::ConnectionI::monitor(const chrono::steady_clock::time_point& now, const ACMConfig& acm)
-{
-    lock_guard lock(_mutex);
-    if (_state != StateActive)
-    {
-        return;
-    }
-    assert(acm.timeout != chrono::seconds::zero());
-
-    //
-    // We send a heartbeat if there was no activity in the last
-    // (timeout / 4) period. Sending a heartbeat sooner than really
-    // needed is safer to ensure that the receiver will receive the
-    // heartbeat in time. Sending the heartbeat if there was no
-    // activity in the last (timeout / 2) period isn't enough since
-    // monitor() is called only every (timeout / 2) period.
-    //
-    // Note that this doesn't imply that we are sending 4 heartbeats
-    // per timeout period because the monitor() method is still only
-    // called every (timeout / 2) period.
-    //
-    if (acm.heartbeat == ACMHeartbeat::HeartbeatAlways ||
-        (acm.heartbeat != ACMHeartbeat::HeartbeatOff && _writeStream.b.empty() &&
-         now >= (_acmLastActivity + chrono::duration_cast<chrono::nanoseconds>(acm.timeout) / 4)))
-    {
-        if (acm.heartbeat != ACMHeartbeat::HeartbeatOnDispatch || _upcallCount > 0)
-        {
-            sendHeartbeatNow();
-        }
-    }
-
-    if (static_cast<int32_t>(_readStream.b.size()) > headerSize || !_writeStream.b.empty())
-    {
-        //
-        // If writing or reading, nothing to do, the connection
-        // timeout will kick-in if writes or reads don't progress.
-        // This check is necessary because the activity timer is
-        // only set when a message is fully read/written.
-        //
-        return;
-    }
-
-    if (acm.close != ACMClose::CloseOff && now >= (_acmLastActivity + acm.timeout))
-    {
-        if (acm.close == ACMClose::CloseOnIdleForceful ||
-            (acm.close != ACMClose::CloseOnIdle && !_asyncRequests.empty()))
-        {
-            //
-            // Close the connection if we didn't receive a heartbeat in
-            // the last period.
-            //
-            setState(StateClosed, make_exception_ptr(ConnectionTimeoutException(__FILE__, __LINE__)));
```

Comment:
As of this PR, ConnectionTimeoutException is not longer created in C++. I will remove it in a follow-up PR.